File: src/ui/src/modules/diet-planner/utils/visibility.ts
// Mirrors DietPlanner.Domain Visibility — who can see a recipe or product.
export const VISIBILITIES = ['Private', 'Household', 'Public'] as const;

export type Visibility = (typeof VISIBILITIES)[number];

export const DEFAULT_VISIBILITY: Visibility = 'Household';

export function toVisibility(value: string): Visibility {
  return VISIBILITIES.find((v) => v === value) ?? DEFAULT_VISIBILITY;
}

File: src/Modules/DietPlanner/DietPlanner.Domain/ValueObjects/Visibility.cs
namespace DietPlanner.Domain.ValueObjects;

/// <summary>
/// Who can see a library item (recipe or product) besides its creator.
/// </summary>
public enum Visibility
{
    /// <summary>Only the creator.</summary>
    Private,

    /// <summary>Members of the creator's household; the default.</summary>
    Household,

    /// <summary>Every user.</summary>
    Public
}
